powerline/lib/vcs/git.py

```diff
@@ -86,6 +87,9 @@ def status(self, path=None):
 	def branch(self):
 		directory = git_directory(self.directory)
 		head = join(directory, 'HEAD')
+		# Resolve symlinks when determining branch to avoid cache conflicts issue#1958
+		if Path(directory).is_symlink():
```

Comment:
I think it would be better to put these kind of functions in https://github.com/powerline/powerline/blob/develop/powerline/lib/path.py

Reply:
Makes sense. I'll modify that.